tests/test_process.py

```diff
@@ -100,6 +101,59 @@ def test_get_s2_metadata_esa_id():
     assert process.get_s2_metadata('S2B_MSIL2A_20200913T151809_N0214_R068_T22WEB_20200913T180530') == {"foo": "bar"}
 
 
+def test_s3_object_is_accessible(s3_stubber):
+    bucket = 'MyBucket'
+    key = 'MyKey'
+
+    params = {
+        'Bucket': bucket,
+        'Key': key
+    }
+
+    s3_stubber.add_response(method='head_object', expected_params=params, service_response={})
+    assert process.s3_object_is_accessible(bucket, key)
+
+    s3_stubber.add_client_error(
+        method='head_object', expected_params=params, service_error_code='404', http_status_code=404
+    )
+    assert not process.s3_object_is_accessible(bucket, key)
+
+    s3_stubber.add_client_error(
+        method='head_object', expected_params=params, service_error_code='403', http_status_code=403
+    )
+    assert not process.s3_object_is_accessible(bucket, key)
+
+    s3_stubber.add_client_error(
+        method='head_object', expected_params=params, service_error_code='500', http_status_code=500
+    )
+    with pytest.raises(botocore.exceptions.ClientError):
+        process.s3_object_is_accessible(bucket, key)
+
+
+def test_get_s2_path(monkeypatch):
+    metadata = {'assets': {'B08': {'href': 's3://senintel-s2-l1c/foo/bar.jp2'}}}
+
+    with monkeypatch.context() as m:
+        m.setattr(process, 's3_object_is_accessible', lambda **kwargs: False)
```

Comment:
would it be much effort to assert the right bucket/key values get passed to `s3_object_is_accessible`?